Translate reference/react-component (#43)
reactjs/de.reactjs.org: content/docs/reference-react-component.md

  - "tips/use-react-with-other-libraries.html"
---

> Diese Seite enthält eine detailierte API-Referenz für die React-Component Klassendefinition. Sie nimmt an, dass du mit den grundlegenden React-Konzepten, wie [Komponenten und Props](/docs/components-and-props.html) und [State und Lifecycle](/docs/state-and-lifecycle.html), vertraut bist. Falls nicht, lese die zuerst.

Reviewer: ```suggestion
Diese Seite enthält eine detaillierte API-Referenz für die React-Component-Klassendefinition. Sie nimmt an, dass du mit den grundlegenden React-Konzepten, wie [Komponenten und Props](/docs/components-and-props.html) und [State und Lifecycle](/docs/state-and-lifecycle.html), vertraut bist. Falls nicht, lies darüber zuerst.
```

[Imperativ Lesen](https://www.duden.de/rechtschreibung/lesen_dozieren_schmoekern#grammatik)

## Überblick {#overview}

> React erlaubt das Definieren von Komponenten als Klassen oder Funktionen. Als Klasse definierte Komponenten bieten aktuell mehr Funktionen, auf welche hier im Detail eingegangen wird. Um eine React Komponente als Klasse zu definieren, muss `React.Component` erweitert werden:

Reviewer: ```suggestion
React erlaubt das Definieren von Komponenten als Klassen oder Funktionen. Als Klasse definierte Komponenten bieten aktuell mehr Funktionen, auf welche hier im Detail eingegangen wird. Um eine React-Komponente als Klasse zu definieren, muss `React.Component` erweitert werden:
```

```js
class Willkommen extends React.Component {
  render() {
    return <h1>Hallo, {this.props.name}</h1>;
  }
}
```

Die einzige Methode, die in einer `React.Component`-Unterklasse implementiert werden *muss*, heißt [`render()`](#render). Alle anderen auf dieser Seite beschriebenen Methoden sind optional.

> **Wir raten streng davon ab, eigene Basiskomponentenklassen zu schreiben** In React Komponenten wird [hauptsächlich Code-Wiederverwendung eher durch Komposition als durch Vererbung erreicht](/docs/composition-vs-inheritance.html).

Reviewer: ```suggestion
**Wir raten streng davon ab, eigene Basiskomponentenklassen zu schreiben.** In React-Komponenten wird [hauptsächlich Code-Wiederverwendung eher durch Komposition als durch Vererbung erreicht](/docs/composition-vs-inheritance.html).
```

>Hinweis:
>

> >React zwingt dich nicht die ES6-Klassensyntax zu verwenden. Falls du sie vermeiden möchtest, kannst du das `create-react-class` Modul oder eine ähnliche eigene Abstraktion verwenden. Sie dir [React ohne ES6](/docs/react-without-es6.html) an, um mehr darüber zu lernen.

Reviewer: ```suggestion
>React zwingt dich nicht die ES6-Klassensyntax zu verwenden. Falls du sie vermeiden möchtest, kannst du das `create-react-class`-Modul oder eine ähnliche eigene Abstraktion verwenden. Sieh dir [React ohne ES6](/docs/react-without-es6.html) an, um mehr darüber zu lernen.
```

### Der Lebenszyklus von Komponenten {#the-component-lifecycle}

> Jede Komponente hat verschiedene "Lifecycle-Methoden", welche du überschreiben kannst um Code zu einem bestimmten Zeitpunkt des Prozesses laufen zu lassen. **Du kannst [dieses Lifecycle-Diagramm](http://projects.wojtekmaj.pl/react-lifecycle-methods-diagram/) als Spickzettel nutzen.** In der Liste unten sind die häufig verwendeten Methoden **fett** hervorgehoben. Die restlichen Methoden existiert für relativ seltene Anwendungsfälle.

Reviewer: ```suggestion
Jede Komponente hat verschiedene "Lifecycle-Methoden", welche du überschreiben kannst, um Code zu einem bestimmten Zeitpunkt des Prozesses laufen zu lassen. **Du kannst [dieses Lifecycle-Diagramm](http://projects.wojtekmaj.pl/react-lifecycle-methods-diagram/) als Spickzettel nutzen.** In der Liste unten sind die häufig verwendeten Methoden **fett** hervorgehoben. Die restlichen Methoden existiert für relativ seltene Anwendungsfälle.
```

>
>- [`UNSAFE_componentWillUpdate()`](#unsafe_componentwillupdate)
>- [`UNSAFE_componentWillReceiveProps()`](#unsafe_componentwillreceiveprops)

#### Das Entfernen {#unmounting}

Diese Methode wird aufgerufen, sobald eine Komponente aus der DOM entfernt wird:

- [**`componentWillUnmount()`**](#componentwillunmount)

#### Fehlerbehandlung {#error-handling}

Diese Methoden werden aufgerufen, sobald während des Rendern, in einer Lifecycle-Methode oder im Konstruktor einer Kindkomponente ein Fehler auftritt.

- [`static getDerivedStateFromError()`](#static-getderivedstatefromerror)
- [`componentDidCatch()`](#componentdidcatch)

### Andere APIs {#other-apis}

Jede Komponente bietet zudem noch andere APIs an:

  - [`setState()`](#setstate)
  - [`forceUpdate()`](#forceupdate)

### Klassenattribute {#class-properties}

  - [`defaultProps`](#defaultprops)
  - [`displayName`](#displayname)

### Instanzattribute {#instance-properties}

  - [`props`](#props)
  - [`state`](#state)

* * *

## Referenz {#reference}

### Meistgenutzte Lifecycle Methods {#commonly-used-lifecycle-methods}

> Die Methoden in diesem Abschnitt decken große Mehrheit der Anwendungsfälle ab, denen du beim Erzeugen von React Komponenten begegnen wirst. **Für eine visuelle Referenz, sieh dir dieses [Lifecycle-Diagramm](http://projects.wojtekmaj.pl/react-lifecycle-methods-diagram/) an.**

Reviewer: ```suggestion
Die Methoden in diesem Abschnitt decken die große Mehrheit der Anwendungsfälle ab, denen du beim Erzeugen von React-Komponenten begegnen wirst. **Für eine visuelle Referenz, sieh dir dieses [Lifecycle-Diagramm](http://projects.wojtekmaj.pl/react-lifecycle-methods-diagram/) an.**
```

### `render()` {#render}

```javascript
render()
```

Die `render()`-Methode ist die einzige erforderliche Methode einer Klassenkomponente.

Beim Aufruf überprüft sie `this.props` und `this.state` und gibt eines der folgenden Typen zurück:

> - **React Elemente.** Typischerweise erzeugt durch [JSX](/docs/introducing-jsx.html). Beispielsweise geben `<div />` und `<MyComponent />` React Elemente zurück, welche React anweisen entsprechend einen DOM-Knoten oder eine andere benutzerdefinierte Komponente zu rendern.

Reviewer: ```suggestion
- **React Elemente.** Typischerweise erzeugt durch [JSX](/docs/introducing-jsx.html). Beispielsweise geben `<div />` und `<MyComponent />` React-Elemente zurück, welche React anweisen, entsprechend einen DOM-Knoten oder eine andere benutzerdefinierte Komponente zu rendern.
```

> - **Arrays und Fragmente.** Erlaubt mehrere Elemente von der render Methode zurück geben zu lassen. Lese die Dokumentation bezüglich [Fragmente](/docs/fragments.html) für mehr Informationen.

Reviewer: ```suggestion
- **Arrays und Fragmente.** Erlaubt mehrere Elemente von der render-Methode zurückgeben zu lassen. Lies die Dokumentation bezüglich [Fragmente](/docs/fragments.html) für mehr Informationen.
```

> - **Portale**. Erlaubt das Rendern von Kindelementen innerhalb eines anderen DOM-Unterbaums. Lese die Dokumentation bezüglich [Portale](/docs/portals.html) für mehr Informationen.

Reviewer: ```suggestion
- **Portale**. Erlaubt das Rendern von Kindelementen innerhalb eines anderen DOM-Unterbaums. Lies die Dokumentation bezüglich [Portale](/docs/portals.html) für mehr Informationen.
```

- **Zeichenketten und Zahlen.** Diese werden als Textknoten in der DOM gerendert.

> - **Booleans or `null`**. Nichts wird gerendert. (Existiert hauptsächlich um das `return test && <Child />`-Muster zu unterstützen, wobei `test` boolean ist.)

Reviewer: ```suggestion
- **Booleans or `null`**. Nichts wird gerendert. (Existiert hauptsächlich um das `return test && <Child />`-Muster zu unterstützen, wobei `test` ein Boolean ist.)
```

Ich konnte nicht herausfinden, ob es das Adjektiv "boolean" im Deutschen gibt und ob es als Adjektiv leicht verständlich ist. Deshalb habe ich es in ein Substantiv geändert.

> Die `render()`-Funktion sollte rein sein, was bedeutet, dass es nicht die State der Komponente verändert, jedesmal das gleiche Ergebnis beim Aufruf liefert und nicht direkt mit dem Browser interagiert.

Reviewer: ```suggestion
Die `render()`-Funktion sollte rein sein, was bedeutet, dass es den State der Komponente nicht verändert, jedesmal das gleiche Ergebnis beim Aufruf liefert und nicht direkt mit dem Browser interagiert.
```
